feat(playwright): support configuring depth for scraping

Nihal-Srivastava05 opened the pull request:
- Add ability to scrape nav bar links directly with param scrape_nav_links (defaults to 0 which means no nav bar links will be scraped). 
- Also added a set to make sure it doesnt scrape the same link twice, this can be overridden using the force=True which will scrape the link regardless of it was previously scraped

Shiti commented:
I think we could support scraping for all the links in the page instead of only the ones in the navbar and use a depth parameter to control/limit how many levels it recursively looks at.

Nihal-Srivastava05 commented:
@Shiti I have updated the agent to scrape all the links instead of just navbar and added a depth as well as external links parameter as discussed. please re-review thanks!

Nihal-Srivastava05 commented:
@Shiti Made all the suggested changes, please re review. Thank you
